### DISCUSSION TITLE
It is recommended to use NetTopologySuite as the cornerstone of the entire project to facilitate the expansion of spatial analysis functionality in the later stage.

### DISCUSSION THREAD

**Discussion by lucinelouhi** [Mapsui/Mapsui#2616] [Ideas]
It is recommended to use NetTopologySuite as the cornerstone of the entire project to facilitate the expansion of spatial analysis functionality in the later stage. I'm not sure if I didn't see it or if it's not in the source code, but I don't think it's appropriate to just stay in one map display. NetTopologySuite itself provides some spatial analysis, but I found that basic interfaces such as IFeature are not consistent with NetTopologySuite. So much so that if spatial topology analysis is needed, a certain transformation is required. It is recommended that some interfaces of the NetTopologySuite library remain consistent. Thank you for your contributions. Maybe I didn't see the complete meaning in the code, please understand.

**Comment by pauldendulk** [comment]
I am not sure if I fully understand you but I will try to answer anyway.

We don't want Mapsui core to depend on NTS. In general it is good to limit you dependencies, it makes the project easier to maintain, and in the future we might want to support other geometry libraries along side NTS. We want to have full control of the core classes of Mapsui.

We advise users to use the rich functionality of NTS. There is a lot that can be used very easily. I think users do not know enough about its functionality. For users it is also good if their NTS logic is not mingled with Mapsui code. We try to keep it use isolated in Mapsui.Nts.

There are downside to this choice, for instance you can't do all the transformations and manipulations you can do on an NTS Point on an MPoint. We accept this downside but we try to make the problem as small as possible. What is possible:
- We could provide and easy conversion with ToMPoint and ToNtsPoint (I guess we already have those). 
- Or you could not use MPoint in your layer but use NTS Point instead. 
- You mention the limitations of IFeature. If you know your IFeature is a GeometryFeature than cast it to a GeometryFeature in your own code and work with the GeometryFeature.Geometry (which is NTS).

Btw, v4 is the first version using NTS. Integration could be improved, but you have to provide more specifics about what you want to do.
